rust-runtime/aws-smithy-runtime/src/client/orchestrator/endpoints.rs

```diff
@@ -144,17 +143,34 @@ fn apply_endpoint(
     endpoint: &Endpoint,
     endpoint_prefix: Option<&EndpointPrefix>,
 ) -> Result<(), BoxError> {
-    let uri: Uri = endpoint.url().parse().map_err(|err| {
-        ResolveEndpointError::from_source("endpoint did not have a valid uri", err)
-    })?;
-
-    apply_endpoint_to_request_uri(request.uri_mut(), &uri, endpoint_prefix).map_err(|err| {
```

Comment:
Is this old apply_endpoint method used anymore, or can it be removed?

Reply:
there is one usage in the ECS provider. I think that can eventually use the endpoint resolver from the orchestrator though.